### PR TITLE
[1.16] Fix possible crash when using rendering regionCache

### LINKED ISSUE #7197: [1.15 & 1.16] Possible crash when using rendering regionCache
**Minecraft Version:** 1.15 & 1.16

**Forge Version:** both latest (1.15.2 - 31.2.33, 1.16.1 - 32.0.98)

**Logs:**
crash log: 
https://gist.github.com/loudwhitenoise/301f5a98f974eeb3eb16fb73d0d1da62 
latest.log: 
https://gist.github.com/loudwhitenoise/f598067f53ec34521cd61f4ece7fa795

**Description of issue:**
CacheLoader [(code link)](https://github.com/MinecraftForge/MinecraftForge/blob/638d1d31deabd8a83cf4b84496549703bc527ab4/src/main/java/net/minecraftforge/client/MinecraftForgeClient.java#L116) may cause crash when returning null. This shouldn't happen in vanilla code (as the cache is filled properly by vanilla), but can happen when rendering fake worlds which are not using ChunkRenderCache properly.

**Proposed solution:**
Make cache value Optional<ChunkRenderCache> and adjust methods accordingly

I can make PR when the final solution is determined

### PR DESCRIPTION
**Pull request by Nightenom**
Fixes #7197 using Optional as cache value wrapper

[This](https://github.com/MinecraftForge/MinecraftForge/pull/7207/files#diff-6b2a01e94e48c3cec54af7d282a1abd0L116) may be null when a chunk is completely empty (air blocks only)

### CODE REVIEW

File: src/main/java/net/minecraftforge/client/MinecraftForgeClient.java [hunk 1]
```diff
@@ -122,10 +123,10 @@ public static void onRebuildChunk(World world, BlockPos position, ChunkRenderCac
         if (cache == null)
             regionCache.invalidate(Pair.of(world, position));
         else
-            regionCache.put(Pair.of(world, position), cache);
+            regionCache.put(Pair.of(world, position), Optional.ofNullable(cache));
     }
 
-    public static ChunkRenderCache getRegionRenderCache(World world, BlockPos pos)
+    public static Optional<ChunkRenderCache> getRegionRenderCache(World world, BlockPos pos)
```
Comment (diesieben07) [inline src/main/java/net/minecraftforge/client/MinecraftForgeClient.java]: This is a binary breaking change. Is there a reason you are not just adding a null check on the call-site?

**Review by diesieben07 — CHANGES_REQUESTED** [review]

File: src/main/java/net/minecraftforge/client/MinecraftForgeClient.java (hunk 1 above)
Comment (Nightenom) [reply src/main/java/net/minecraftforge/client/MinecraftForgeClient.java]: Guave Cache does not allow null values as mentioned in linked issue (#7197)

Comment (diesieben07) [reply src/main/java/net/minecraftforge/client/MinecraftForgeClient.java]: That doesn't mean this method needs to return an Optional.

Comment (Nightenom) [reply src/main/java/net/minecraftforge/client/MinecraftForgeClient.java]: Surely I can rename this to getRegionRenderCacheOptional and make the old method link it

Comment (diesieben07) [reply src/main/java/net/minecraftforge/client/MinecraftForgeClient.java]: Or just not change the method at all.

Comment (Nightenom) [reply src/main/java/net/minecraftforge/client/MinecraftForgeClient.java]: Done

**Comment by Nightenom** [comment]
@diesieben07 anything else what needs to be addressed? I'm gonna make backport once this is ready

File: src/main/java/net/minecraftforge/client/MinecraftForgeClient.java [hunk 2]
```diff
@@ -115,10 +116,15 @@ public static void onRebuildChunk(World world, BlockPos position, ChunkRenderCac
         if (cache == null)
             regionCache.invalidate(Pair.of(world, position));
         else
-            regionCache.put(Pair.of(world, position), cache);
+            regionCache.put(Pair.of(world, position), Optional.ofNullable(cache));
```
Comment (marchermans) [inline src/main/java/net/minecraftforge/client/MinecraftForgeClient.java]: You can use a simple `Optional.of(...)` here since `cache` can not be null.

**Review by marchermans — APPROVED** [review]
One minor nitpicking then this looks good.

**Pull request merged**